### PR TITLE
 Bulk insert of failed receipts using query macro

### LINKED ISSUE #273: [Feat.Req] Bulk insert of failed receipts
**Problem statement**
We have a loop that inserts all the failed receipts, but it's way more efficient if we insert via bulk.

### PR DESCRIPTION
**Pull request by taslimmuhammed**
Fixes #273 
After running the commands and successful compilation:-
cargo clippy --all-targets --all-features 
cargo fmt 
cargo sqlx prepare --workspace -- --all-targets

### CODE REVIEW

File: tap-agent/src/agent/sender_allocation.rs [hunk 1]
```diff
@@ -696,31 +704,43 @@ impl SenderAllocationState {
...
+            values.push(BigDecimal::from(BigInt::from(receipt.message.value)));
         }
+        sqlx::query!(
+            r#"INSERT INTO scalar_tap_receipts_invalid (
+                signer_address,
+                signature,
+                allocation_id,
+                timestamp_ns,
+                nonce,
+                value
```
Comment (gusinacio) [inline tap-agent/src/agent/sender_allocation.rs:721]: It's missing here `error_log`

**Review by gusinacio — CHANGES_REQUESTED** [review]
Invalid receipts have an extra field where we are able to store the reason why the receipt failed.

This new implementation is missing it. Let me know if you need help to add them.

File: tap-agent/src/agent/sender_allocation.rs (hunk 1 above)
Comment (taslimmuhammed) [reply tap-agent/src/agent/sender_allocation.rs:721]: did you meant this line 
debug!(
                    "Receipt for allocation {} and signer {} failed reason: {}",
                    allocation_id.encode_hex(),
                    receipt_signer.encode_hex(),
                    receipt_error
            );

Comment (gusinacio) [reply tap-agent/src/agent/sender_allocation.rs:721]: We want to store the error logs for all receipts as well.

Take a look at the older version, we were inserting `error_log` in the old sql.

**Pull request closed without merging**